src/function/scalar/map/map.cpp

```diff
@@ -2,11 +2,58 @@
 #include "duckdb/common/string_util.hpp"
 #include "duckdb/parser/expression/bound_expression.hpp"
 #include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/function/aggregate/nested_functions.hpp"
 #include "duckdb/common/types/data_chunk.hpp"
 #include "duckdb/common/pair.hpp"
+#include "duckdb/common/types/value_map.hpp"
 
 namespace duckdb {
 
+// TODO: this doesn't recursively verify maps if maps are nested
+void VerifyMap(Vector &map, idx_t count, const SelectionVector &sel) {
```

Comment:
Can we split this function into a function that returns an `enum class MapInvalidReason : uint8_t { MAP_VALID, MAP_NULL_KEY_LIST, MAP_NULL_KEY, MAP_NOT_UNIQUE };` and a function that throws the `InvalidInputExceptions`?

in `Vector::Verify` we want to throw an `InternalException` if the map is not valid, and not an `InvalidInputException`. We want to do something like:

```cpp
D_ASSERT(VerifyMapInternal(map, count, sel) == MapInvalidReason::MAP_VALID);
```

Reply:
Done 👍 
I turned VerifyMap into a static method of Vector, that just asserts that it's valid
CheckMapValidity now returns the enum
and I've added two functions that handle the errors
One in arrow_conversion.cpp and one in map.cpp

Reply:
Most of the ones in Arrow conversion should never be hit, because the only thing we don't share is the unique requirement, but I added them for completeness